### DISCUSSION TITLE
REQUEST :: Need help modifying a plugin compatible with Joomla 4x

### DISCUSSION THREAD

**Discussion by XenyFord** [joomla/joomla-cms#38051] [Q&A] [answered]
Dear Intellects & Bright Sparks, Good morning :)

I've been using a popup login plugin on my Joomla3.x website, and I recently upgraded my website to joomla 4.x got to know that the popuplogin plugin is not compatible with joomla4x.

Although I'm not very familiar with PHP and coding, I'm tried to make some changes to make it work on joomla 4x website, but still nothing worked, so my humble request to the members here if anyone can help me make the plugin compatible with Joomla 4x, please. :pray: :pray: :pray:

_Please find the the attached Joomla 3.x plugin below;_

[plg_system_loginpopup_v1.0.1_j30.zip](https://github.com/joomla/joomla-cms/files/8899132/plg_system_loginpopup_v1.0.1_j30.zip)

Thank you in advance :pray: :pray: :pray:

**Comment by brianteeman** [comment]
You really should contact extstore the makers of that plugin

**Reply by XenyFord** [reply]
YES! Sir, I contacted them many times for the same. But, came to know that the maker, extstore has been closed its operations since a very long ago. My humble request to you or any of the members here please help me If possible. :pray: :pray: :pray:

**Reply by XenyFord** [reply]
But, If possible, please help me and don't get me wrong anyway.. :pray: :pray: :pray::pray: :pray:

**Reply by ghost** [reply]
Not developing anymore

**Comment by richard67** [comment]
So what can you do? You could
- go through this document and check if that extensions uses some of the mentioned things, e.g. removed classes or calls like "isAdmin" or "isSite" and replace them by the right calls for J4 https://docs.joomla.org/Potential_backward_compatibility_issues_in_Joomla_4, or
- go through this great tutorial by @astridx https://blog.astrid-guenther.de/en/der-weg-zu-joomla4-erweiterungen/ and modify that extensions accordingly, or rewrite it partly or completely.
- If you want to pay a professional developer doing that for you and you are on Facebook, you could join https://www.facebook.com/groups/joomlajobsgroup and ask there if someone wants to do that work.

**Comment by brianteeman** [comment]
Should have a working version for you shortly

**Reply by XenyFord** [reply]
Millions of heartfelt thanks to you in advance sir!

**Reply by brianteeman** [reply]
Sorry for the delay in replying. After digging into the plugin and seeing exactly how it works its not as simple as I had thought becuause j4 has much more in the login

**Reply by XenyFord** [reply]
Hi Sir, 
It's working as expected. Really a lots of heartfelt thanks to you...Much grateful to you.. :)

**Comment by brianteeman** [comment, marked as answer]
Instead of converting the plugin I have taken a different approach and released LoginModal https://github.com/brianteeman/loginmodal/releases/tag/1.0.0

Unlike your original plugin there is no need to replicate all the login module settings in the plugin etc. This plugin uses the core joomla login module.

**Reply by XenyFord** [reply]
Dear Mr. Teenman,
Millions of heartfelt thanks to you...Much grateful to you sir :+1: :pray::pray: :pray: :pray: :pray:

**Reply by XenyFord** [reply]
Mr. Teenman, Good morning :)

I hope you are doing great, sir!

I've enabled the Modal Login module on a website, and it is working fine on all the pages except a page where the EventBooking component is enabled. It has just disappeared from creating a popup div. I request you to please be so kind to help me rectify this problem.

Website: https://astro-remedies.com/events

Thank you in advance for your help!

![ERROR](https://github.com/joomla/joomla-cms/assets/89290310/80fdb765-3d54-4159-bac7-347dd51cd4c9)

---Working Modal on other pages ---

![WORKING POPUP](https://github.com/joomla/joomla-cms/assets/89290310/0fca5088-5d79-4841-a48e-810f8c8a1a0f)
